services/QuillLMS/client/app/bundles/PremiumHub/components/usage_snapshots/snapshotCount.tsx

```diff
@@ -0,0 +1,134 @@
+  selectedTimeframe: string;
+  adminId: number;
+  customTimeframeStart?: any;
+  customTimeframeEnd?: any;
+  comingSoon?: boolean;
+  passedCount?: number;
+  passedChange?: number;
+  passedChangeDirection?: 'negative'|'positive';
+  singularLabel?: string;
+}
+
+const SnapshotCount = ({ label, size, queryKey, comingSoon, searchCount, selectedGrades, selectedSchoolIds, selectedTimeframe, customTimeframeStart, customTimeframeEnd, adminId, passedCount, passedChange, passedChangeDirection, singularLabel, }: SnapshotCountProps) => {
+  const [count, setCount] = React.useState(passedCount || null)
+  const [change, setChange] = React.useState(passedChange || 0)
+  const [changeDirection, setChangeDirection] = React.useState(passedChangeDirection || null)
+  const [loading, setLoading] = React.useState(false)
+
+  React.useEffect(() => {
+    if (comingSoon) { return }
+
+    resetToDefault()
+
+    getData()
+  }, [searchCount])
+
+  function resetToDefault() {
+    setCount(passedCount || null)
+    setChangeDirection(passedChangeDirection || null)
+    setChange(passedChange || 0)
+  }
+
+  function getData() {
+    initializePusher()
+
+    const searchParams = {
+      query: queryKey,
+      timeframe: selectedTimeframe,
+      timeframe_custom_start: customTimeframeStart,
+      timeframe_custom_end: customTimeframeEnd,
+      school_ids: selectedSchoolIds,
+      grades: selectedGrades
+    }
+
+    const requestUrl = queryString.stringifyUrl({ url: '/snapshots/count', query: searchParams }, { arrayFormat: 'bracket' })
+
+    requestGet(`${requestUrl}`, (body) => {
+      if (!body.hasOwnProperty('current')) {
+        setLoading(true)
+      } else {
+        const { previous, current, } = body
+
+        const roundedCurrent = Math.round(current || 0)
+
+        setCount(roundedCurrent)
+
+        if (!previous) {
+          setChangeDirection(POSITIVE)
+          return
+        }
+
+        const roundedPrevious = Math.round(previous || 0)
+
+        const changeTotal = Math.round(((roundedCurrent - roundedPrevious) / (roundedPrevious || 1)) * 100)
+        setChange(Math.abs(changeTotal))
+        setChangeDirection(changeTotal > 0 ? POSITIVE : NEGATIVE)
+        setLoading(false)
+      }
+    })
+  }
+
+  function initializePusher() {
+    const pusher = new Pusher(process.env.PUSHER_KEY, { encrypted: true, });
+    const channel = pusher.subscribe(String(adminId));
+    channel.bind('admin-snapshot-count-cached', (body) => {
+      const { message, } = body
+
+      const queryKeysAreEqual = message.query === queryKey
+      const timeframesAreEqual = message.timeframe === selectedTimeframe
+      const schoolIdsAreEqual = _.isEqual(message.school_ids, selectedSchoolIds.map(id => String(id)))
+      const gradesAreEqual =  _.isEqual(message.grades, selectedGrades.map(grade => String(grade)))
+
+      if (queryKeysAreEqual && timeframesAreEqual && schoolIdsAreEqual && gradesAreEqual) {
+        getData()
+      }
+    });
+  };
```

Comment:
I really like the check to make sure that the Pusher filters match the current filters before re-fetching data.  Very nice.